docs/general/ddl/data-types.rst

```diff
@@ -991,12 +993,13 @@ inserted into it, but otherwise ignore them.
 .. NOTE::
 
    ``Ignored`` objects should be mainly used for storing and fetching.
-   Filtering by and ordering on them is possible but very performance
-   intensive. ``Ignored`` objects are a *black box* for the storage engine, so
-   the filtering/ordering is done using an expensive table scan and a
-   filter/order function outside of the storage engine. Using ``ignored``
-   objects for grouping or aggregations is not possible at all and will result
-   in an exception or ``NULL`` value if used with excplicit casts.
+   Filtering by and ordering on any undefined subcolumn is possible but very
+   performance intensive. Undefined subcolumns of ``ignored`` objects are a
+   *black box* for the storage engine, so the filtering/ordering is done using
+   an expensive table scan and a filter/order function outside of the storage
+   engine. Using undefined subcolumns of ``ignored`` objects for grouping or
+   aggregations is not possible and will result in an exception unless explicit
+   casted.
```

Comment:
Also a suggestion here to elaborate a bit on why it is slower and to distinguish between the ordering and filtering case, because ordering is not *that* much more expensive compared to the difference when using these columns in the WHERE clause. I also noticed that the part about not being able to order or use aggregations is incorrect:

```suggestion
   Given that dynamically added sub-columns of an ``ignored`` objects are not indexed, filter operations on these columns cannot utilize the index and instead a value lookup is performed for each matching row. This can be mitigated by combining a filter using the ``AND`` clause with other predicates on indexed columns.

   Futhermore, values for dynamically added sub-columns of an ``ignored`` objects aren't stored in a column store, which means that ordering on these columns or using them with aggregates is also slower than using the same operations on regular columns. For some operations it may also be necessary to add an explicit type cast because there is no type information available in the schema. An example::


   cr> SELECT arbitrary(payload['value']::text) FROM metrics;


Given that it is possible have values of different types within the same sub-column of an ignored objects, aggregations may fail at runtime::


   cr> SELECT sum(payload['value']::bigint) FROM metrics;
```

Reply:
The `arbitrary` example is probably not deterministic, so maybe a filter or so will be needed in addition - or you could come up with a better example?

Reply:
> For some operations it may also be necessary to add an explicit type cast because there is no type information available in the schema

Afaik the explicit type cast is *always*  needed despite simple selecting the column. (functions, aggs, grouping, ordering, filtering, etc.)

Reply:
> Afaik the explicit type cast is _always_ needed despite simple selecting the column. (functions, aggs, grouping, ordering, filtering, etc.)

I think due to the type inference it can be omitted if a function overload is restrictive enough. The following works:

```
create table tbl (id text primary key, payload object (ignored));
insert into tbl (id, payload) values ('1', {x=10});
insert into tbl (id, payload) values ('2', {x=10});
refresh table tbl;
select sum(payload['x']) from tbl;
```